mirascope/core/anthropic/_utils/_handle_stream.py

```diff
@@ -59,31 +60,55 @@ def _handle_chunk(
     if chunk.type == "content_block_delta" and chunk.delta.type == "input_json_delta":
         buffer += chunk.delta.partial_json
 
+        # Return partial tool if enabled
+        if partial_tools and current_tool_type:
+            partial_tool_call = ToolUseBlock(
+                id=current_tool_call.id,
+                input=buffer,
+                name=current_tool_call.name,
+                type="tool_use",
+            )
+            partial_tool = current_tool_type.from_tool_call(partial_tool_call)
+            if hasattr(chunk.delta, "partial_json"):
```

Comment:
why do we need this check here? we access `chunk.delta.partial_json` above just fine?